clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -463,17 +465,23 @@ void pollSelectionKeys(Set<SelectionKey> selectionKeys,
 
             } catch (Exception e) {
                 String desc = channel.socketDescription();
-                if (e instanceof IOException)
-                    log.debug("Connection with {} disconnected", desc, e);
-                else
-                    log.warn("Unexpected error from {}; closing connection", desc, e);
+                logException(e, desc);
                 close(channel, true);
             } finally {
                 maybeRecordTimePerConnection(channel, channelStartTimeNanos);
             }
         }
     }
 
+    private void logException(Exception e, String desc) {
+        if (e instanceof SSLException)
+            log.error("Connection with {} disconnected", desc, e);
```

Comment:
Are all `SSLExceptions` fatal? Also, we should probably update the log message.

Reply:
AFAIK, SSLHandshakeException, SSLKeyException, SSLPeerUnverifiedException are fatal exceptions, SSLProtocolException is not common.  We can check with @rajinisivaram  

Updated logs.

Reply:
On the broker-side this is not fatal, but typically caused by a mis-configured client. For clients, it is typically fatal, but could sometimes just be a clock-mismatch where a retry could succeed.

Reply:
@ijuma @rajinisivaram  any suggestion on this PR?